src/util/sortByKeyOrder.js

```diff
@@ -1,7 +1,7 @@
 function findIndex(arr, err) {
   let idx = Infinity;
   arr.some((key, ii) => {
-    if (err.path.indexOf(key) !== -1) {
+    if (new RegExp(`^${err.path}$`).test(key)) {
```

Comment:
won't this break for complex paths? like: `'foo.bar['baz']'`

Reply:
All the tests have passed. How can I test this? My yup usage is very basic with formik.

Reply:
I'd adjust your test to use a nested object. 

```
object({
  foo: object({
    bar: string()
  })
})
```

Reply:
Done. Take a look at it, pls.

Reply:
Hi, @jquense. Do you have any other consideration about this?

Reply:
Sorry for the ping, @jquense, but is there something I can do to get this merged?

Reply:
sorry for forgetting this. I don't think this is right still `new RegExp(`^${err.path}$`)` will create a regular expression that matches slightly wrong things, because paths contain regex tokens that would need to be escaped, in particular `.`. However, the regex here isn't doing anything other than `a === err.path`, which is to say it matches the entire string exactly (or would if the path was escaped). 

This also won't work if the nested object needs it's keys sorted:

```js
object({
  nested: object({
    foo: string().when('bar', () => string().min(5)),
    bar: object({
      foo: string().required(),
    }),
    barfoo: string().required(),
  }),
});
```

When findIndex runs on `nested`'s children the field array will be `['foo', 'bar', 'barfoo'] but the `error.path` will start with `nested` e.g. it will be 'nested.foo` which won't match any paths. Honestly i'm not sure how to solve this, but it seems like i'd be more complicated than is worth it, but definitely open to ideas!